### ISSUE TITLE
Code fix for CS0029, CS0266 "Change return type ..." should not be suggested for overriding members and interface implementations

### ISSUE THREAD

**Issue by JosefPihrt** [dotnet/roslyn#17357]
**Version Used**: Visual Studio 2017 RC 15.0.0-RC4+26206.0

**Code to Reproduce**:
```csharp
namespace My
{
    public interface IFoo
    {
        string Bar();
    }

    public class Foo : IFoo
    {
        public string Bar()
        {
            // CS0029 is reported here
            // should not suggest changing return type to 'bool'
            return false;
        }
    }

    public class Base
    {
        public virtual string Bar()
        {
            return null;
        }
    }

    public class Derived : Base
    {
        public override string Bar()
        {
            // CS0266 is reported here
            // should not suggest changing return type to 'object'
            return default(object);
        }
    }
}
```

**Comment by Pilchie** [comment]
This fix is not provided by Roslyn.

**Comment by JosefPihrt** [comment]
Oh, I did not realize that I have installed third-party package that provides fixes for "CS...." diagnostics. My apologies for creating this irrelevant issue.
